### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by RGBCube**

### CODE REVIEW

File: config.dev.template.toml [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
+    00000000000000000,
+]
+prefixes = [
+    "!",
+    "?",
+]
+testGuildIds = [
+    00000000000000000,
+    00000000000000000,
+]
```
Comment (Xirado) [inline config.dev.template.toml:13]: what the fuck?

Comment (Xirado) [reply config.dev.template.toml:13]: For token you can just do `<BOT_TOKEN>` or something and for the ids `1234` is fine

Comment (RGBCube) [reply config.dev.template.toml:13]: Alright

File: src/main/kotlin/io/github/gasbarrel/Config.kt [hunk 2]
```diff
@@ -3,42 +3,56 @@ package io.github.gasbarrel
...
+import kotlin.io.path.Path
+import kotlin.io.path.exists
 import kotlin.io.path.readText
 
+val config: Config by lazy {
+    val configPath = Config.folder.resolve("config.toml")
+
+    Config.logger.info("Loading configuration at $configPath")
+
+    return@lazy Toml.decodeFromString(configPath.readText())
```
Comment (Xirado) [inline src/main/kotlin/io/github/gasbarrel/Config.kt:19]:
```suggestion
    Toml.decodeFromString(configPath.readText())
```

**Review by Xirado — CHANGES_REQUESTED** [review]

**Comment by RGBCube** [comment]
The main big behavioural change is the handling of configuration:
- If `dev-config/` exists, use that dir and set isDev to true, if not, use `config/` dir
- Use `dev-data/` or `data/` depeding on isDev
- logback.xml should be in both config dirs

**Comment by RGBCube** [comment]
> > The main big behavioural change is the handling of configuration:
> > ```
> > * If `dev-config/` exists, use that dir and set isDev to true, if not, use `config/` dir
> > 
> > * Use `dev-data/` or `data/` depeding on isDev
> > 
> > * logback.xml should be in both config dirs
> >   And all of this is relative to the working directory, no relative to jar only in prod nonsense
> > ```
> 
> You should keep it so it's relative to the jar's location, this way you can have multiple versions of the same bot (1 directory = 1 bot) without accidentally mixing up the data by using the same working directory

So it should *always* be relative to the jar location?

File: src/main/kotlin/io/github/gasbarrel/Environment.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package io.github.gasbarrel
+
+import java.nio.file.Path
+import kotlin.io.path.Path
+import kotlin.io.path.exists
+import kotlin.io.path.toPath
+
+object Environment {
+    /**
+     * The mode is determined by checking whether if the
```
Comment (Xirado) [inline src/main/kotlin/io/github/gasbarrel/Environment.kt:10]:
```suggestion
     * The mode is determined by checking if the
```

**Review by Xirado — CHANGES_REQUESTED** [review]

**Review by Xirado — APPROVED** [review]

**Pull request merged**